### PR TITLE
Wrap diagnostics

### PR DESCRIPTION
**Pull request by dangunter**
Proposed changes:
Wrap diagnostics functionality in idaes-pse so it can be consumed by the UI layer (as JSON, not arbitrarily formatted text)

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by dangunter** [comment]
All the tests that matter pass. I don't know what's going on with the documentation build, but this PR doesn't touch the docs so I say ignore it and just base approval or change requests on the code files.

**Comment by dangunter** [comment]
@CopyDemon could you review

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
 --editable .[dev]
 idaes-pse @ git+https://github.com/IDAES/idaes-pse@main
-pytest-icdiff >= 0.7  # readable dict diffs for test_flowsheet and others
\ No newline at end of file
+pydantic~=2.0
+fastapi
```
Comment (CopyDemon) [inline requirements-dev.txt:4]: should we add version numbers for both Fastapi and Uvicorn

File: idaes_ui/fv/app.py
Comment (CopyDemon) [inline idaes_ui/fv/app.py:1]: in the idaes when call m.fs.visualize("fs_name") won't enable this script, and it still use the old server.

**Review by CopyDemon — APPROVED** [review]

File: requirements-dev.txt (hunk 1 above)
Comment (dangunter) [reply requirements-dev.txt:4]: not unless we have to

**Pull request merged**